spark-jobs/src/main/scala/za/co/absa/enceladus/common/version/SparkVersionGuard.scala

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.common.version
+
+import org.slf4j.{Logger, LoggerFactory}
+import za.co.absa.commons.version.Version
+import za.co.absa.commons.version.impl.SemVer20Impl.SemanticVersion
+import za.co.absa.enceladus.common.SparkCompatibility
+
+object SparkVersionGuard {
+
+  /**
+   * Populates the version guard with the defaults from [[za.co.absa.enceladus.common.SparkCompatibility]]
+   */
+  def fromDefaultSparkCompatibilitySettings: SparkVersionGuard =
+    SparkVersionGuard(SparkCompatibility.minSparkVersionIncluded, SparkCompatibility.maxSparkVersionExcluded)
+}
+
+/**
+ * Setup Spark version guard with allowed min & max sem-ver version.
+ *
+ * @param minVersionInclusive lowest acceptable spark version (may be non-final)
+ * @param maxVersionExclusive version supremum - first discouraged spark version (this and higher version usage issues a warning) if not None
+ */
+case class SparkVersionGuard(minVersionInclusive: SemanticVersion, maxVersionExclusive: Option[SemanticVersion])
+                            (implicit log:Logger = LoggerFactory.getLogger(SparkVersionGuard.getClass)) {
```

Comment:
I am just confused/unsure where is this logger coming from?

Reply:
I am not completely sure if I follow, but let me try:
if a Logger is given (explicitly or implicitly), it is used, otherwise, this default implementation is used. In the actual sparkjobs, the default impl is used; In test, both the default or explicit logger are used (as you commented below).

If you are wondering if this "parameter group" needs to be implicit, it does necessarily does not; it could have also been just an optional parameter -- currently, the implicit param passing is not used. But it can be if we wanted so, so I included it.

Reply:
Sorry, I was blind. I failed to see the default value for the parameter. 🙄